### PR TITLE
bpo-8256: Fixed possible failing or crashing input()

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
if attributes "encoding" or "errors" of sys.stdin or sys.stdout
are not set or are not strings.

### CODE REVIEW

**Review by vstinner — APPROVED** [review]
LGTM, but I didn't test the change.

File: Python/bltinmodule.c [hunk 1]
```diff
@@ -1928,10 +1928,12 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
 
         stdin_encoding = _PyObject_GetAttrId(fin, &PyId_encoding);
         stdin_errors = _PyObject_GetAttrId(fin, &PyId_errors);
-        if (!stdin_encoding || !stdin_errors)
+        if (!stdin_encoding || !stdin_errors ||
+            !PyUnicode_Check(stdin_encoding) ||
+            !PyUnicode_Check(stdin_errors))
```
Comment (vadmium) [inline Python/bltinmodule.c]: Isn’t this already covered by the PyUnicode_AsUTF8 (aka _PyUnicode_AsString in 3.5) error checking below?

Also, I would use a double indent here to distinguish the “if” condition from the body.

Comment (serhiy-storchaka) [reply Python/bltinmodule.c]: This is the main point of this change. If `PyUnicode_AsUTF8()` is failed due to memory error or unpaired surrogates, this is an error. But if *encoding* or *error* are not strings (but `None` for example), use the fallback.

File: Python/bltinmodule.c [hunk 2]
```diff
@@ -2011,7 +2015,9 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
         return NULL;
     }
 
+fallback:
     /* Fallback if we're not interactive */
+    PyErr_Clear();
```
Comment (serhiy-storchaka) [inline Python/bltinmodule.c]: Hmm, need to clear references here.

**Pull request merged**